src/tools/wasm-opt.cpp

```diff
@@ -149,7 +149,11 @@ int main(int argc, const char* argv[]) {
     ModuleReader reader;
     reader.setDebug(options.debug);
     try {
-      reader.read(options.extra["infile"], wasm, inputSourceMapFilename);
+      if (options.extra["infile"].size()) {
```

Comment:
can we push this check into the ModuleReader logic? Then we wouldn't need to modify the other commandline tools too.